src/palimpzest/query/optimizer/rules.py

```diff
@@ -237,6 +254,8 @@ def substitute(cls, logical_expression: LogicalExpression, **physical_op_params)
             generated_fields=logical_expression.generated_fields,
             group_id=logical_expression.group_id,
         )
```

Comment:
should we also do the `deduped_physical_expressions` pattern that is used below here?